### PR TITLE
feat: check savedSearch query field

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **FEATURE**

This PR resolves [FX-2967]

### Description
As an admin with the saved search feature flag enabled
_When I_ apply filters on the artist artwork screen
_A query_ is executed that checks if I have a saved search with those filters

##### Acceptance criteria:
* Every time a user with the saved search flag applies filters, a query is executed that checks if that user has an existing saved search with those filters
* End result of this ticket could just be logging out the result of that query

#### Demo

https://user-images.githubusercontent.com/3513494/121049697-9e650180-c7c0-11eb-9c11-e93e46673b2e.mp4

https://user-images.githubusercontent.com/3513494/121050503-4c70ab80-c7c1-11eb-8feb-832316e97550.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[FX-2967]: https://artsyproduct.atlassian.net/browse/FX-2967

### CODE REVIEW

File: src/lib/Components/Artist/ArtistArtworks/SavedSearchBanner.tsx [hunk 1]
```diff
@@ -34,3 +115,48 @@ export const SavedSearchBanner: React.FC<SavedSearchBannerProps> = ({ enabled, l
...
+    <QueryRenderer<SavedSearchBannerQuery>
+      environment={defaultEnvironment}
+      query={graphql`
+        query SavedSearchBannerQuery($criteria: SearchCriteriaAttributes!) {
+          me {
+            ...SavedSearchBanner_me @arguments(criteria: $criteria)
+          }
+        }
+      `}
+      render={({ props, error }) => (
```
Comment (dimatretyak) [inline src/lib/Components/Artist/ArtistArtworks/SavedSearchBanner.tsx:149]: Should the banner be displayed if an error occurred during the request?

Comment (dblandin) [reply src/lib/Components/Artist/ArtistArtworks/SavedSearchBanner.tsx:149]: Sometimes you can have an error but still have the props you need to render.

There's this pattern which will either console.log the error or send it to Sentry, which seems reasonable:

https://github.com/artsy/eigen/blob/master/src/lib/Scenes/Search/AutosuggestResults.tsx#L206-L211

**Comment by dblandin** [comment]
> > We should eventually add specs that exercise both GraphQL mutations
> 
> @dblandin If it is not difficult, can you describe in more detail what is meant by this?

Sure! I'm thinking of something like this which sets up some mock mutation data, does something in the UI (like tap the enable/disable) button, and asserts on some kind of response like a toast appearing or the banner switching to the other state.

There should be some examples in Eigen but here's one example in Force that I'm familiar with: https://github.com/artsy/force/blob/8aa26fc8215cbb679ba7d14f81d25d23fc6337e9/src/v2/Components/UserSettings/TwoFactorAuthentication/__tests__/TwoFactorAuthentication.jest.tsx#L124

**Comment by dimatretyak** [comment]
@dblandin

* [createSavedSearch mutation spec](https://artsyproduct.atlassian.net/browse/FX-3006)
* [deleteSavedSearch mutation spec](https://artsyproduct.atlassian.net/browse/FX-3006)

**Pull request merged**